packages/web-app-external/src/App.vue

```diff
@@ -88,73 +80,69 @@ export default {
       })
     },
     appName() {
-      return this.$route.params.appName
-    },
-    fileId() {
-      return this.$route.params.fileId
+      return this.$route.query.app
     }
   },
+  mounted() {
+    const appNameTitle = this.appName ? `${this.appName} - ` : ''
+    document.title = `${this.currentFileContext.fileName} - ${appNameTitle}${this.configuration.currentTheme.general.name}`
+  },
   async created() {
-    await unauthenticatedUserReady(this.$router, this.$store)
-
     this.loading = true
-
-    // build headers with respect to the actual auth situation
-    const { 'public-token': publicToken } = this.$route.query
-    const publicLinkPassword = this.publicLinkPassword
-    const accessToken = this.getToken
-    const headers = {
-      'X-Requested-With': 'XMLHttpRequest',
-      ...(publicToken && { 'public-token': publicToken }),
-      ...(publicLinkPassword && {
-        Authorization:
-          'Basic ' + Buffer.from(['public', publicLinkPassword].join(':')).toString('base64')
-      }),
-      ...(accessToken && {
-        Authorization: 'Bearer ' + accessToken
-      })
-    }
-
-    // fetch iframe params for app and file
-    const configUrl = this.configuration.server
-    const appOpenUrl = this.capabilities.files.app_providers[0].open_url.replace(/^\/+/, '')
-    const url =
-      configUrl +
-      appOpenUrl +
-      `?file_id=${this.fileId}` +
-      (this.appName ? `&app_name=${this.appName}` : '')
-
-    const response = await fetch(url, {
-      method: 'POST',
-      headers
-    })
-
-    if (response.status !== 200) {
-      this.errorMessage = response.message
+    try {
+      const filePath = this.currentFileContext.path
+      const fileInfo = await this.getFileInfoResource(filePath)
```

Comment:
We would already have the file id in the route query. Feels unnecessary to fetch the whole file info again just to extract the id and don't do anything else with it.